extensions/brew/src/components/runInTerminal.ts

```diff
@@ -1,10 +1,17 @@
 import { closeMainWindow } from "@raycast/api";
 import { runAppleScript } from "run-applescript";
+import { preferences } from "../preferences";
 
 export function runCommandInTerminal(command: string): void {
   runAppleScript(`
-    tell application "Terminal"
-      do script "${command}"
+    tell application "${preferences.useIterm2 ? "iTerm" : "Terminal"}"
+      ${
+        preferences.useIterm2
+          ? "set newWindow to create window with default profile command"
+          : "do shell script \"open -a 'Terminal'\"\ndo script"
+      } "bash -c '${command}; read -n 1 -s -r -p \\"Press any key to exit - will not quit\\" ; echo' ; exit" ${
+    preferences.useIterm2 ? "" : "in selected tab of the front window"
```

Comment:
The apple script is getting a bit messy and hard to follow here. I'd suggest breaking out into two separate invocations, one for Terminal and one for iTerm.